tests/lib/pkgdb.sh

```diff
@@ -166,13 +192,65 @@ distro_auto_remove_packages() {
         fedora-*)
             dnf -q -y autoremove
             ;;
+        opensuse-*)
+            ;;
         *)
             echo "ERROR: Unsupported distribution '$SPREAD_SYSTEM'"
             exit 1
             ;;
     esac
 }
 
+distro_query_package_info() {
+    case "$SPREAD_SYSTEM" in
+        ubuntu-*|debian-*)
+            apt-cache policy "$1"
+            ;;
+        fedora-*)
+            dnf info "$1"
+            ;;
+        opensuse-*)
+            zypper info "$1"
+            ;;
+    esac
+}
+
+distro_install_build_snapd(){
+    if [ "$SRU_VALIDATION" = "1" ]; then
+        apt install -y snapd
+        cp /etc/apt/sources.list sources.list.back
+        echo "deb http://archive.ubuntu.com/ubuntu/ $(lsb_release -c -s)-proposed restricted main multiverse universe" | tee /etc/apt/sources.list -a
+        apt update
+        apt install -y --only-upgrade snapd
+        mv sources.list.back /etc/apt/sources.list
+        apt update
+    else
+        packages=
+        case "$SPREAD_SYSTEM" in
+            ubuntu-*|debian-*)
+                packages="${GOHOME}/snapd_*.deb"
+                ;;
+            fedora-*|opensuse-*)
+                packages="${GOHOME}/snap-confine*.rpm ${GOPATH}/snapd*.rpm"
```

Comment:
We could but don't have to, Zypper wont fail interestingly. However let me split these.